### ISSUE TITLE
Enhance a MarianMT pretrained model from HuggingFace with more training data

### ISSUE THREAD

**Issue by stelmath** [huggingface/transformers#6985]
# ❓ Questions & Help

## Details

**A link to original question on the forum/Stack Overflow**: https://stackoverflow.com/questions/63774619/enhance-a-marianmt-pretrained-model-from-huggingface-with-more-training-data

**Comment by raghuch** [comment]
( I have had the same issue, and tried some steps to finetune. I am reproducing my answer to the above stackoverflow question here again)
Have you tried the finetune.sh script shown [here](https://github.com/huggingface/transformers/blob/master/examples/seq2seq/finetune.sh)? In addition to the short list of CLI flags listed there, you could try adding:

```
--src_lang "en" \
--tgt_lang "de" \
--num_train_epochs 400 \
--warmup_steps 20 \
--train_batch_size 32 \
--eval_batch_size 32 \
--data_dir "/data/dir" \
--output_dir "/path/to/store/model/etc" \
--cache_dir "/path/for/misc/files" \
--max_source_length 128 \
--max_target_length 128 \
--val_max_target_length 128 \
--test_max_target_length 128 \
--model_name_or_path "</path/to/pretrained>"
```

where the "/path/to/pretrained" could be either a local path on your machine or MarianMT model (Opus-en-de or equivalent). The "data/dir" has a "train.source" and "train.target" for the source & target languages, such that line number x of the target is a translation of line x in the source (and same with "val.source" and "val.target"). 

Note: The gradients blew up when I used the "fp16" flag, so I had removed it. Also, you might want to check on the "val_check_interval", "check_val_every_n_epoch", and probably check [this issue](https://github.com/huggingface/transformers/issues/3447) on how to save multiple checkpoints.
